### ISSUE TITLE
oF static lib product named differently when built in Xcode than from make

### ISSUE THREAD

**Issue by ghost** [openframeworks/openFrameworks#2882]
I've run into this a few times and been confused but when building the oF static lib, Xcode uses a different build product name than when building from `make` :frowning:

Xcode: `libs/openFrameworksCompiled/lib/osx/openFrameworksDebug.a`
make: `libs/openFrameworksCompiled/lib/osx/libopenFrameworksDebug.a`

Should these product names be the same?

**Comment by pizthewiz** [comment]
Still a bummer that `make` and the Xcode project create and use different OF static libs.

**Comment by flvi0** [comment]
I would like to give this a try in order to make the oF cmake config for external projects not having to account for this, as mentioned in #4430. 

@pizthewiz @ofTheo Regarding the ordering, the ld manpage says "Both dynamic and static libraries are searched as they appear on the command line.", and also
```
-Ldir       Add dir to the list of directories in which to search for libraries.  Directories specified with -L are searched in the order they appear on the command line and before the default search path.
```
So I think we should be fine if the project/application gets the libdir ordering right.

Regarding the invasiveness towards users, we might solve this via symlinking the old name to the new one for the time being, announcing deprecation of that library naming and maybe get rid of it with the 0.9 release? I don't know if there's any scheme on this, but I guess an upcoming major release might be a point in time? What do you think?

**Comment by flvi0** [comment]
Ok got it renamed and the post-build symlinking worked out. Why is the lib deployed in the bin/ folder actually?

**Comment by ofTheo** [comment]
I agree with @pizthewiz here - we shouldn't rename the Xcode / IDE library names, but fix it so make generates openFrameworks.a and openFrameworksDebug.a 

There is far less potential for causing new issues with this approach.  
And although the way we are not using the lib prefix on OS X is non-standard, I think if we are going to not have libFreetype.a but FreeType.a for 3rd party libs we should also have it be the same for openFrameworks.

**Comment by flvi0** [comment]
@ofTheo I can't actually read out of @pizthewiz's posts that he prefers that non-standard naming scheme, and I must say that tend to disagree. Reading the docs (since I'm kinda new to MacOS X) libraries are clearly expected to be prefixed with lib on a mac system, see http://www.manpages.info/macosx/ld.1.html about the -lx option. With GNU ld there's the possibility so use the -soname option to tweak this, I have as of yet not found something for Mach ld which does the same. I presume it exists, but I don't see the reason to jump through extra hoops (and have every project depending on oF required to do so as well) in order to use a non-standard naming scheme. What are your motivations behind that? Where do you think this would cause less new issues? I would actually expect just the opposite.

**Comment by flv0** [comment]
Since I got no answer here and in the pull request, can I assume there is no longer any interest in this? I might then close the open pull requests.

**Comment by bakercp** [comment]
Just wanted to confirm that this is still an issue 10.12 xcode 8.3.1, master branch.
